### PR TITLE
[test plan] Macsec rekeying

### PR DESCRIPTION
**Pull request by parmarkj**
### Description of PR

This test is to verify macsec rekey occurs after re-configuring the rekey period.

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205
- [ ] 202305

#### What is the motivation for this PR?
To add a new test case.

#### How did you do it?
Please refer to the readme file.

#### How did you verify/test it?
Verified on T2.

#### Supported testbed topology if it's a new test case?
T2

### CODE REVIEW

**Comment by parmarkj** [comment]
mssonicbld, apologies. Code has been updated to pass pre-commit. Thanks

File: tests/macsec/test_rekeying.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    pytest.mark.macsec_required,
+    pytest.mark.topology("any")
+]
+
+profile_name = "256_XPN_SCI"
+new_rekey_period = 180
+
+
+@pytest.fixture(scope='module')
+def setup(duthost, macsec_nbrhosts, ctrl_links):
```
Comment (wenyiz2021) [inline tests/macsec/test_rekeying.py:26]: @parmarkj can you add T2 support for this testcase?
here in the setup you could use `duthosts` and `enum_rand_one_per_hwsku_frontend_hostname`

**Comment by wenyiz2021** [comment]
@judyjoseph for viz
the way to check if macsec rekey happened by `grep` may not be correct.
the swss.rec file might contains old rekey keyword even if no rekey happened the last time

File: tests/macsec/test_rekeying.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    logger.info('Setup_info: {}'.format(setup_info))
+
+    yield setup_info
+
+
+def test_rekeying(duthost, setup, ctrl_links):
+    macsec_profile = setup['macsec_profile']
+
+    logger.info("Remove macsec and reconfigure with rekey period set to {}".format(macsec_profile['rekey_period']))
+    setup_macsec_configuration(duthost, ctrl_links, macsec_profile['name'],
```
Comment (BYGX-wcr) [inline tests/macsec/test_rekeying.py:62]: In the test objective, you wrote that the code should disable the macsec port first. However, setup_macsec_configuration does not explicitly disable the macsec port. Can you please explain the inconsistency here?

File: tests/macsec/test_rekeying.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                               macsec_profile['cipher_suite'],
+                               macsec_profile['primary_cak'],
+                               macsec_profile['primary_ckn'],
+                               macsec_profile['policy'],
+                               macsec_profile['send_sci'],
+                               macsec_profile['rekey_period'])
+
+    logger.info("Wait for rekey to occur")
+    # Wait for few rekeys to occur
+    time.sleep(macsec_profile['rekey_period'])
```
Comment (BYGX-wcr) [inline tests/macsec/test_rekeying.py:73]: The programs only waits for a single rekey_period here before it checks the log. How robust it is against possible logging latency?

File: docs/testplan/Macsec_Rekeying.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
+- [Overview](#overview)
+    - [Scope](#scope)
+    - [Testbed](#testbed)
+- [Setup configuration](#setup-configuration)
+- [Test cases](#test-cases)
+
+# Test name
+
+Macsec Rekeying
```
Comment (judyjoseph) [inline docs/testplan/Macsec_Rekeying.md:9]: @parmarkj Thanks for this test -- just wanted to point out there is already a test in macsec testsuite to cover rekey scenario. https://github.com/sonic-net/sonic-mgmt/blob/fa5ab2566c9e30a6cbdaab87345eadd9ae63e7b2/tests/macsec/test_controlplane.py#L65

Can you check if that test can be enhanced ?

Comment (parmarkj) [reply docs/testplan/Macsec_Rekeying.md:9]: closing PR as test already covered

**Pull request closed without merging**